src/puppetlabs/puppetdb/mq.clj

```diff
@@ -164,10 +164,15 @@
    (let [^TextMessage text-message message]
      (.getText text-message))
    (instance? javax.jms.BytesMessage message)
-   (let [^BytesMessage bytes-message message]
-     (.readUTF8 bytes-message))
+   (let [^BytesMessage bytes-message message
+         len (.getBodyLength message)
+         buf (byte-array len)]
+     (assert (= len (.readBytes bytes-message buf)))
```

Comment:
Should throw something better than an assertion error if this check fails; at least it should include a message.